include/boost/json/detail/basic_parser.hpp

```diff
@@ -138,63 +129,109 @@ class basic_parser
     bool more_; // false for final buffer
     bool complete_ = false; // true on complete parse
     bool is_key_;
+    const char* end_;
     parse_options opt_;
 
     inline static bool is_control(char c) noexcept;
     inline static char hex_digit(char c) noexcept;
+    
     inline void reserve();
-    inline void suspend(state st);
-    inline void suspend(state st, number const& num);
-    inline bool skip_white(const_stream& cs);
-
-    result syntax_error(const_stream&);
+    inline const char* canary();
+
+    BOOST_NOINLINE
+    inline
+    const char*
+    propagate(state st);
```

Comment:
This is not descriptive enough

Reply:
IMO it's the least misleading. I'll add docs saying what it actually does.